Mathlib/Topology/Algebra/ContinuousMonoidHom.lean

```diff
@@ -108,6 +108,50 @@ instance instMonoidHomClass : MonoidHomClass (ContinuousMonoidHom A B) A B where
 instance instContinuousMapClass : ContinuousMapClass (ContinuousMonoidHom A B) A B where
   map_continuous f := f.continuous_toFun
 
+section
+
+variable {A B : Type*} [Monoid A] [Monoid B] [TopologicalSpace A] [TopologicalSpace B]
+  {F : Type*} [FunLike F A B]
+
+/-- Turn an element of a type `F` satisfying `MonoidHomClass F A B` and `ContinuousMapClass F A B`
+into a`ContinuousMonoidHom`. This is declared as the default coercion from `F` to
+`ContinuousMonoidHom A B`. -/
+@[to_additive (attr := coe) "Turn an element of a type `F` satisfying
+`AddMonoidHomClass F A B` and `ContinuousMapClass F A B` into a`ContinuousAddMonoidHom`.
+This is declared as the default coercion from `F` to `ContinuousAddMonoidHom A B`."]
+def toContinuousMonoidHom [MonoidHomClass F A B] [ContinuousMapClass F A B] (f : F) :
+    ContinuousMonoidHom A B :=
+  { MonoidHomClass.toMonoidHom f with }
+
+/--Any type satisfying `MonoidHomClass` and `ContinuousMapClass` can be cast into
+`ContinuousMnoidHom` via `ContinuousMonoidHom.toContinuousMonoidHom`.-/
+@[to_additive "Any type satisfying `AddMonoidHomClass` and `ContinuousMapClass` can be cast into
+`ContinuousAddMonoidHom` via `ContinuousAddMonoidHom.toContinuousAddMonoidHom`."]
+instance [MonoidHomClass F A B] [ContinuousMapClass F A B] : CoeTC F (ContinuousMonoidHom A B) :=
+  ⟨ContinuousMonoidHom.toContinuousMonoidHom⟩
+
+@[to_additive (attr := simp)]
+lemma coe_fun [MonoidHomClass F A B] [ContinuousMapClass F A B] (f : F) :
```

Comment:
elsewhere in the library this lemma is called `coe_coe`. You can use that if you prefer.
```suggestion
@[to_additive (attr := simp, norm_cast)]
lemma coe_toContinuousMonoidHom [MonoidHomClass F A B] [ContinuousMapClass F A B] (f : F) :
```